cmake/public/utils.cmake

```diff
@@ -570,3 +570,26 @@ function(torch_update_find_cuda_flags)
                     "    CUDA_NVCC_FLAGS_MINSIZEREL     = ${FLAGS_MINSIZEREL}")
   endif()
 endfunction()
+
+##############################################################################
+# CHeck if given flag is supported and append it to provided outputvar
+# Also define HAS_UPPER_CASE_FLAG_NAME variable
+# Usage:
+#   append_cxx_flag_if_supported("-Werror" CMAKE_CXX_FLAGS)
+function(append_cxx_flag_if_supported flag outputvar)
+    string(TOUPPER "HAS${flag}" _FLAG_NAME)
+    string(REGEX REPLACE "[=-]" "_" _FLAG_NAME "${_FLAG_NAME}")
+    check_cxx_compiler_flag("${flag}" ${_FLAG_NAME})
```

Comment:
What exactly does this? Wouldn't a singular call to `check_cxx_compiler_flag` with a value of "1" for `${_FLAG_NAME}` do the right thing here?

Reply:
This is explain in function description: FLAG_NAME Is always global, so it must be unique.
I.e. if the same var is used to check for two different flag 2nd check would not even be run and FOO would be true
```
check_cxx_compiler_flag("-Werror" FOO)
check_cxx_compiler_flags("-bogus" FOO)
```

Reply:
CMAKE does have a concept of local vars, but `check_[c|cxx]_compiler_flag` is special and global by design, so that repeated calls to that function returns in noops rather than actual compiler invocations